resolwe/flow/models/entity.py

```diff
@@ -133,6 +127,12 @@ class Meta(BaseModel.Meta):
             ("owner_relation", "Is owner of the relation"),
         )
 
+    def save(self, *args, **kwargs):
+        """Lowercase the ``category`` field and save the model."""
+        self.category = self.category.lower()
```

Comment:
Why do we want the category lower case? We want to keep the original case, but make uniqueness and search invariant of the case. The reason we want the original case is that the category really acts like name (or label) in the sense that it will be displayed in many places in UI. And if the category is abbreviation of something it should be all caps, maybe they want just the first letter capitalized, maybe the category will be a name of something and all words will have the first letter capitalized... Image the user name a relation (sets the category), and then this relation has a different case in UI. Maybe category will be _Homo sapiens vs Mus musculus study_.  Biologists would be pissed if we mess up the case there.